File: SelectionSort/src/br/com/ip/sort/SelectionSort.java
package br.com.ip.sort;

public class SelectionSort {

	public static void main(String[] args) {
		
		int [] array  = {1, 5, 2, 3};
		
		System.out.println("Disorderly");
		for(int i: array) {
			System.out.println(i);
		}
		
		selectionSort(array);
		
		System.out.println("Orderly");
		for(int i: array) {
			System.out.println(i);
		}

	}
	
	public static void selectionSort(int[] arr) {
	    
		int n = arr.length;
	    
	    for (int i = 0; i < n-1; i++) {
	        int minIdx = i;
	        
	        for (int j = i+1; j < n; j++) {
	            if (arr[j] < arr[minIdx]) {
	                minIdx = j;
	            }
	        }
	        	
	        int temp = arr[i];
	        arr[i] = arr[minIdx];
	        arr[minIdx] = temp;
	    }
	}


}

File: SelectionSort/bin/br/com/ip/sort/SelectionSort.py

def selection_sort(arr):
    n = len(arr)
    
    for i in range(n):
        min_idx = i
        for j in range(i+1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j

        arr[i], arr[min_idx] = arr[min_idx], arr[i]


listNumbers = [1,5,2,3,4]
selection_sort(listNumbers)

print(listNumbers)

